apps/src/lib/client/eth_bridge/validator_set.rs

```diff
@@ -44,3 +51,80 @@ pub async fn query_validator_set_args(args: args::ActiveValidatorSet) {
 
     println!("0x{}", HEXLOWER.encode(encoded_validator_set_args.as_ref()));
 }
+
+/// Relay a validator set update, signed off for a given epoch.
+pub async fn relay_validator_set_update(args: args::ValidatorSetUpdateRelay) {
+    let nam_client = HttpClient::new(args.query.ledger_address).unwrap();
+
+    let epoch_to_relay = if let Some(epoch) = args.epoch {
+        epoch
+    } else {
+        RPC.shell().epoch(&nam_client).await.unwrap().next()
+    };
+    let shell = RPC.shell().eth_bridge();
+    let encoded_proof_fut =
+        shell.read_valset_upd_proof(&nam_client, &epoch_to_relay);
+
+    let bridge_current_epoch = Epoch(epoch_to_relay.0.saturating_sub(2));
+    let shell = RPC.shell().eth_bridge();
+    let encoded_validator_set_args_fut =
+        shell.read_active_valset(&nam_client, &bridge_current_epoch);
+
+    let shell = RPC.shell().eth_bridge();
+    let governance_address_fut = shell.read_governance_contract(&nam_client);
+
+    let (encoded_proof, encoded_validator_set_args, governance_contract) =
+        futures::try_join!(
+            encoded_proof_fut,
+            encoded_validator_set_args_fut,
+            governance_address_fut
+        )
+        .unwrap();
+
+    let (bridge_hash, gov_hash, signatures): (
+        [u8; 32],
+        [u8; 32],
+        Vec<Signature>,
+    ) = abi_decode_struct(encoded_proof);
```

Comment:
I seems like we are doing lots of unnecessary rounds of serialization.

Reply:
the RPC method could return a non-abi encoded data response. right now it borsh serializes the abi encoding of the data. CLI commands requesting a proof are kind of obsolete now, but if we wanted to keep them anyway (so users could potentially implement shell scripts around this or smth?), they could be responsible for returning the ABI encoding of the data queried from our RPC endpoints

Reply:
I think it makes more sense to borsh serialize the raw struct and down the road, if some script needs abi encoding, it can call abi encoding on the raw struct itself. I don't think that's an unreasonable burden to put on people implementing their own scripts. 

I just feel like we have a lot of artifact code that I would like to clear out. This damn repo is complicated enough as it is.

Reply:
I feel the same. I'll work on this later though, right now it's out of the scope of this PR